hazelcast/src/test/java/com/hazelcast/executor/SingleNodeTest.java

```diff
@@ -0,0 +1,346 @@
+        assertTrue(latch.await(2, TimeUnit.SECONDS));
+    }
+
+    @Test(timeout = 10000)
+    public void testNestedExecution() throws Exception {
+        Callable<String> task = new NestedExecutorTask();
+        executor.submit(task).get();
+    }
+
+    @Test
+    public void testMultipleFutureGets() throws Exception {
+        Callable<String> task = new BasicTestTask();
+        Future<String> future = executor.submit(task);
+        String s1 = future.get();
+        assertEquals(s1, BasicTestTask.RESULT);
+        assertTrue(future.isDone());
+        String s2 = future.get();
+        assertEquals(s2, BasicTestTask.RESULT);
+        assertTrue(future.isDone());
+        String s3 = future.get();
+        assertEquals(s3, BasicTestTask.RESULT);
+        assertTrue(future.isDone());
+        String s4 = future.get();
+        assertEquals(s4, BasicTestTask.RESULT);
+        assertTrue(future.isDone());
+    }
+
+    @Test
+    public void testInvokeAll() throws Exception {
+        assertFalse(executor.isShutdown());
+        // Only one task
+        ArrayList<Callable<String>> tasks = new ArrayList<Callable<String>>();
+        tasks.add(new BasicTestTask());
+        List<Future<String>> futures = executor.invokeAll(tasks);
+        assertEquals(futures.size(), 1);
+        assertEquals(futures.get(0).get(), BasicTestTask.RESULT);
+        // More tasks
+        tasks.clear();
+        for (int i = 0; i < 1000; i++) {
+            tasks.add(new BasicTestTask());
+        }
+        futures = executor.invokeAll(tasks);
+        assertEquals(futures.size(), 1000);
+        for (int i = 0; i < 1000; i++) {
+            assertEquals(futures.get(i).get(), BasicTestTask.RESULT);
+        }
+    }
+
+    @Test
+    public void testInvokeAllTimeoutCancelled() throws Exception {
+        assertFalse(executor.isShutdown());
+        final List<? extends Callable<Boolean>> singleTask = Collections.singletonList(new SleepingTask(0));
+        List<Future<Boolean>> futures = executor.invokeAll(singleTask, 5, TimeUnit.SECONDS);
+        assertEquals(futures.size(), 1);
+        assertEquals(futures.get(0).get(), Boolean.TRUE);
+
+        final List<Callable<Boolean>> tasks = new ArrayList<Callable<Boolean>>();
+        for (int i = 0; i < 1000; i++) {
+            tasks.add(new SleepingTask(i < 2 ? 0 : 20));
+        }
+        futures = executor.invokeAll(tasks, 5, TimeUnit.SECONDS);
+        assertEquals(futures.size(), 1000);
+        for (int i = 0; i < 1000; i++) {
+            if (i < 2) {
+                assertEquals(futures.get(i).get(), Boolean.TRUE);
+            } else {
+                //noinspection EmptyCatchBlock
+                try {
+                    futures.get(i).get();
+                    fail();
+                } catch (CancellationException expected) { }
+            }
+        }
+    }
+
+    @Test
+    public void testInvokeAllTimeoutSuccess() throws Exception {
+        assertFalse(executor.isShutdown());
```

Comment:
you have already a test `testShutdownBehaviour()` and you have the same assertion. If you are not sure about this assertion, i guess you have to write new test for this.